src/test/scala/edu/ie3/osmogrid/lv/coordinator/LvCoordinatorSpec.scala

```diff
@@ -0,0 +1,329 @@
+      )
+    val inputDataProviderAdapter =
+      asynchronousTestKit.createTestProbe[InputDataProvider.Response](
+        "InputDataProviderAdapter"
+      )
+    val regionCoordinatorAdapter =
+      asynchronousTestKit.createTestProbe[LvRegionCoordinator.Response](
+        "InputDataProviderAdapter"
+      )
+    val msgAdapters = MessageAdapters(
+      inputDataProviderAdapter.ref,
+      regionCoordinatorAdapter.ref
+    )
+
+    "being initialized" should {
+
+      "register the correct amount of message adapters" in {
+        val idleTestKit = BehaviorTestKit(
+          coordinator.LvCoordinator(
+            cfg,
+            inputDataProvider.ref,
+            lvCoordinatorAdapter.ref
+          )
+        )
+
+        idleTestKit.expectEffectType[
+          MessageAdapter[InputDataProvider.Response, coordinator.Request]
+        ]
+        idleTestKit.expectEffectType[
+          MessageAdapter[LvRegionCoordinator.Response, coordinator.Request]
+        ]
+      }
+    }
+
+    "being idle" should {
+      "terminate if asked to do so" in {
+        val idleTestKit = BehaviorTestKit(
+          coordinator.LvCoordinator(
+            cfg,
+            inputDataProvider.ref,
+            lvCoordinatorAdapter.ref
+          )
+        )
+        idleTestKit.run(coordinator.Terminate)
+
+        idleTestKit.logEntries() should contain only CapturedLogEvent(
+          Level.INFO,
+          "Got request to terminate."
+        )
+        idleTestKit.isAlive shouldBe false
+      }
+
+      val idleTestKit = BehaviorTestKit(
+        coordinator.LvCoordinator(
+          cfg,
+          inputDataProvider.ref,
+          lvCoordinatorAdapter.ref
+        )
+      )
+      "ask for osm and asset information upon request" in {
+        idleTestKit.run(coordinator.ReqLvGrids)
+
+        /* Receive exactly two messages, that are requests for OSM and asset data */
+        inputDataProvider.receiveMessages(2).forall {
+          case _: InputDataProvider.ReqOsm =>
+            true
+          case _: InputDataProvider.ReqAssetTypes =>
+            true
+          case _ => false
+        } shouldBe true
+      }
+    }
+
+    "awaiting input data" should {
+      "terminate if asked to do so" in {
+        val awaitingTestKit = BehaviorTestKit(
+          LvCoordinator invokePrivate PrivateMethod[Behavior[
+            coordinator.Request
+          ]](
+            Symbol(
+              "awaitInputData"
+            )
+          )(
+            AwaitingData.empty(
+              IdleData(
+                cfg,
+                inputDataProvider.ref,
+                lvCoordinatorAdapter.ref,
+                msgAdapters
+              )
+            )
+          )
+        )
+
+        awaitingTestKit.run(coordinator.Terminate)
+
+        awaitingTestKit.logEntries() should contain only CapturedLogEvent(
+          Level.INFO,
+          "Got request to terminate."
+        )
+        awaitingTestKit.isAlive shouldBe false
+      }
```

Comment:
Can we get issues here that the log and alive check are evaluated before the actor receives the message and shuts down?

Reply:
AFAIK not, because the test kit is made for synchronous testing. It uses only one thread and thereby, I would not see any chance, that the order of events gets mixed up.